src/Foundatio/Caching/CacheClientHelpers.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Threading.Tasks;
+
+namespace Foundatio.Caching
+{
+    public static class CacheClientHelpers
+    {
+        public static async Task<CacheValue<T>> GetOrAddAsync<T>(this ICacheClient client, string key, Func<T> addFunc,
+            TimeSpan? expiresIn = null)
```

Comment:
Yeah, its best if it doesn't have to create another async state machine behind the scenes just to throw an error :), unless anyone disagrees :)

https://github.com/exceptionless/Foundatio/blob/master/src/Foundatio/Caching/InMemoryCacheClient.cs#L99